### PR TITLE
prefix api routes with /api/v1

### LINKED ISSUE #92: Consider Prefixing routes with /api or a version
We don’t need to do this, and certainly not immediately, but I think it’s worth considering. 

Using a /api prefix (or anything unique really) gives us the flexibility to host static pages and the API on the same server without worrying about conflicts. It doesn’t seem unreasonable to deploy the social front end to the same server. 

Similarly, a version identifier allows us to more easily make big changes going forward without worrying about breaking a current client’s use. That could be as simple as not having to update the social site at the same time as deploying a backend feature but it could also be about more easily working with research projects. 

Versions could be part of the URL or an HTTP header, but I usually fine putting them towards the front of the URL to be easy, I.e. /v1/projects....

Combined together we could preview API routes with /api/v1, but simply using /v1 as a prefix would work too because it’s not likely to conflict with something else. 

Anyways, like I said, this isn’t urgent but something to consider since it’s a change that’s easier to make when there’s less usage.

### PR DESCRIPTION
**Pull request by cycomachead**
This prefixes only the API routes. Fixes #92 
I've wrapped the prefix in `()` for now. It means that the prefix is optional.

I think we should deploy this in three steps:
* deploy this tweak
* update all cloud.js locations to use the url
* remove the `()` and make the api prefix required.

We can technically try to sync deployments, but this removes that overhead.

### CODE REVIEW

**Comment by cycomachead** [comment]
The intention is to deploy this and the required Snap<em>!</em> change at the same time as when migrating to a new cloud server. This should be staged with the existing cloud.js before then just to ensure backwards compatibility with existing installs.

File: api.lua [hunk 1]
```diff
@@ -44,19 +44,26 @@ require 'passwords'
 
 -- API Endpoints
 -- =============
+-- Wrap all API requests in the same prefix.
+-- TODO: Make /api/v1 be required by removing ().
+-- Do this after Snap! is updated.
+local function api_route(name, path, fn)
```
Comment (cycomachead) [inline api.lua]: Note to self:

We should replace fn with tbl, and then insert the OPTIONS into that table. Each method should be wrapped in capture_errors automatically.

**Comment by cycomachead** [comment]
@bromagosa What do you think of this?

This should standardize the API a bit and gets rid on some boilerplate code in each path.

**Comment by bromagosa** [comment]
Definitely cleaner, but we'll need `cloud.js` to comply.

Or we could add a 404 fallback that tries to prefix a non-found route to the current prefix. There are APIs out there that do that.

**Comment by bromagosa** [comment]
I haven't deployed this yet. I need to leave now and I prefer to do it tomorrow in my early morning, when we both are awake.

**Comment by bromagosa** [comment]
This seems to work great in staging. Deploying it now :)

**Pull request merged**